### ISSUE TITLE
[Bug] Matern Kernel does not work

### ISSUE THREAD

**Issue by umt0** [cornellius-gp/gpytorch#1817]
# 🐛 Bug

Using **gpytorch.kernels.MaternKernel** , the results are not the same as for:

- the formal result https://en.wikipedia.org/wiki/Mat%C3%A9rn_covariance_function
- the Matern kernel from sklearn ( **sklearn.gaussian_process.kernels** ), which coincides with formal result.

## To reproduce

** Code snippet to reproduce **
```python
import gpytorch.kernels
import torch
import numpy as np
import matplotlib.pyplot as plt
from sklearn.gaussian_process.kernels import Matern

nus = np.array([0.5,1.5,2.5])

for nu in nus:
    lengthscale = 1

    K = gpytorch.kernels.MaternKernel(lengthscale=lengthscale,nu = nu)
    x1 = torch.arange(0,1,0.01)
    x2 = torch.zeros(1)
    K_eval = K(x1,x2)
    K_vec = torch.zeros(len(x1))
    K_vec[:] = K_eval[:,0]

    print(K_vec.size())

    K_sk = Matern(length_scale=lengthscale, nu=nu)

    x1 = np.arange(0,1,0.01)
    x1arr = np.zeros((len(x1),1))
    x1arr[:,0] = x1

    x2 = np.zeros(1)
    x2arr = np.zeros((len(x2),1))
    x2arr[:,0] = x2

    K_call = K_sk(x1arr,x2arr)

    K_vec1 = np.zeros(len(x1))
    K_vec1[:] = K_call[:,0]

    print(np.shape(K_vec1))


    plt.xlabel(r'$x$', fontsize=16)
    plt.ylabel(r'$K(x,0)$', fontsize=16, rotation =90)
    plt.plot(x1,K_vec.detach().numpy(),":",label=r"gpytorch, $\nu=%.1f$"%nu)
    plt.plot(x1,K_vec1,"-",label=r"sklearn, $\nu=%.1f$"%nu)
    if nu ==2.5:
        plt.plot(x1,(1+np.sqrt(5)*x1+(5/3)*x1**2)*np.exp(-np.sqrt(5)*x1),"--",label=r"truth, $\nu=%.1f$"%nu,color="black")
    elif nu==1.5:
        plt.plot(x1,(1+np.sqrt(3)*x1)*np.exp(-np.sqrt(3)*x1),"--",label=r"truth, $\nu=%.1f$"%nu,color="black")
    elif nu==0.5:
        plt.plot(x1,(1)*np.exp(-np.sqrt(1)*x1),"--",label=r"truth, $\nu=%.1f$"%nu,color="black")    
        
legend = plt.legend(loc='best', ncol=1,shadow=False,prop={'size': 12}, bbox_to_anchor=(1.05, 1))
plt.show()
```

## Expected Behavior

- The label "truth" stands for formal result: https://en.wikipedia.org/wiki/Mat%C3%A9rn_covariance_function
- The label "sklearn" stands for sklearn Matern kernel
- The label "gpytorch" stands for gpytorch Matern kernel

I am using \nu = 1/2, 3/2 and 5/2. 
I am using lengthscale =1.

**The sklearn captures the formal result, while gpytorch not.**

![image](https://user-images.githubusercontent.com/63372632/141157658-63f7f91b-498a-4f88-b90c-c7617e655fa7.png)

## System information

**Please complete the following information:**
-  I am using gpytorch-1.5.1

**Comment by gpleiss** [comment]
Are you sure that the lengthscale is 1? You have a typo in your code: `lenghtscale=lenghtscale`. Also, to set the lengthscale, you need to type:

`K.initialize(lengthscale=1)`.

This looks to me like gpytorch and sklearn are using different lengthscale values (the default in GPyTorch is something around 0.68).

**Comment by umt0** [comment]
You are completely right! If I add the following:

`   K.initialize(lengthscale=lengthscale)
    K.initialize(nu=nu)`

Both kernels give the right result. Thank you a lot!
